pkg/controller/build/recovery.go

```diff
@@ -87,8 +87,6 @@ func (action *errorRecoveryAction) Handle(ctx context.Context, build *v1alpha1.B
 		return nil, nil
 	}
 
-	build.Status = v1alpha1.BuildStatus{}
-	build.Status.Phase = v1alpha1.BuildPhaseInitialization
```

Comment:
I think resetting the phase is needed otherwise the build won'r be triggered again, isn't it ?

Reply:
Yep - Sorry, that wasn't meant to be removed. I misinterpreted your earlier comments. That's fixed now. 

However, when testing on top of the latest changes, if I provide an invalid dependency to `kamel run`, the build is actually somehow successful. That's definitely not the behaviour when I force HEAD back to cb988c7543ed9a17a44c53c35c50cee4a9ef19c5, the build fails and the recovery loop kicks in as expected.

Reply:
interesting, do you have time to investigate ?

Reply:
Yeah, I'll take a look tomorrow.

Reply:
I think we're good to merge this now that we have #807 fixed.